avalon/tools/workfiles/app.py

```diff
@@ -186,201 +240,261 @@ def refresh(self):
-        self._messagebox.setWindowTitle("Unsaved Changes!")
-        self._messagebox.setText(
-            "There are unsaved changes to the current file."
-            "\nDo you want to save the changes?"
-        )
-        self._messagebox.setStandardButtons(
-            self._messagebox.Yes | self._messagebox.No |
-            self._messagebox.Cancel
-        )
-        result = self._messagebox.exec_()
 
-        if result == self._messagebox.Yes:
-            return True
-        elif result == self._messagebox.No:
-            return False
-        else:
-            return None
+class FilesWidget(QtWidgets.QWidget):
+    """A widget displaying files that allows to save"""
+    def __init__(self, parent=None):
+        super(FilesWidget, self).__init__(parent=parent)
 
-    def open(self, filepath):
+        # Setup
+        self._asset = None
+        self._task = None
+        self.root = None
+        self.host = api.registered_host()
+
+        widgets = {
+            "list": QtWidgets.QListWidget(),
+            "duplicate": QtWidgets.QPushButton("Duplicate"),
+            "open": QtWidgets.QPushButton("Open"),
+            "browse": QtWidgets.QPushButton("Browse"),
+            #"currentFile": QtWidgets.QLabel(),
+            "save": QtWidgets.QPushButton("Save As")
+        }
+
+        # Build buttons widget for files widget
+        buttons = QtWidgets.QWidget()
+        layout = QtWidgets.QHBoxLayout(buttons)
+        layout.setContentsMargins(0, 0, 0, 0)
+        #layout.addWidget(widgets["duplicate"])
+        layout.addWidget(widgets["open"])
+        layout.addWidget(widgets["browse"])
+
+        # Build files widgets
+        layout = QtWidgets.QVBoxLayout(self)
+        layout.setContentsMargins(0, 0, 0, 0)
+        layout.addWidget(widgets["list"])
+        layout.addWidget(buttons)
+
+        separator = QtWidgets.QFrame()
+        separator.setFrameShape(QtWidgets.QFrame.HLine)
+        separator.setFrameShadow(QtWidgets.QFrame.Plain)
+        layout.addWidget(separator)
+
+        #layout.addWidget(widgets["currentFile"])
+        layout.addWidget(widgets["save"])
+
+        widgets["list"].doubleClicked.connect(self.on_open_pressed)
+        widgets["duplicate"].pressed.connect(self.on_duplicate_pressed)
+        widgets["open"].pressed.connect(self.on_open_pressed)
+        widgets["browse"].pressed.connect(self.on_browse_pressed)
+        widgets["save"].pressed.connect(self.on_save_as_pressed)
+
+        self.widgets = widgets
+
+    def set_asset_task(self, asset, task):
+        self._asset = asset
+        self._task = task
+
+    def _get_session(self):
+        """Return a modified session for the current asset and task"""
+
+        session = api.Session.copy()
+        # todo: expose this in the API?
+        changes = pipeline.compute_session_changes(session,
+                                                   asset=self._asset,
+                                                   task=self._task)
```

Comment:
Should we expose this in API? Or is call to `pipeline.xx` fine inside the tools? I suppose it's fine?